genesis/parsing/accountsParser.go

```diff
@@ -302,77 +303,182 @@ func (ap *accountsParser) GetInitialAccountsForDelegated(addressBytes []byte) []
+	txsPoolPerShard map[uint32]*indexer.Pool,
+) {
+	for _, id := range indexingData {
+		for txHash, tx := range id.GetScrsTxs() {
+			senderShardID := shardCoordinator.ComputeId(tx.GetSndAddr())
+			receiverShardID := shardCoordinator.ComputeId(tx.GetRcvAddr())
+
+			scrsTx, ok := tx.(*scrData.SmartContractResult)
+			if !ok {
+				continue
+			}
+			scrsTx.GasLimit = uint64(0)
+
+			txsPoolPerShard[senderShardID].Scrs[txHash] = scrsTx
+			txsPoolPerShard[receiverShardID].Scrs[txHash] = scrsTx
+		}
+	}
+}
+
+func (ap *accountsParser) setTxsPoolAndMiniBlocks(
+	shardCoordinator sharding.Coordinator,
+	allTxs []coreData.TransactionHandler,
+	txsPoolPerShard map[uint32]*indexer.Pool,
+	miniBlocks []*block.MiniBlock,
+) error {
+	for _, txHandler := range allTxs {
+		receiverShardID := shardCoordinator.ComputeId(txHandler.GetRcvAddr())
+		var senderShardID uint32
+		if bytes.Equal(txHandler.GetSndAddr(), ap.minterAddressBytes) {
+			senderShardID = receiverShardID
+		} else {
+			senderShardID = shardCoordinator.ComputeId(txHandler.GetSndAddr())
+		}
+
+		txHash, err := core.CalculateHash(ap.marshalizer, ap.hasher, txHandler)
 		if err != nil {
-			return nil, nil, err
+			return err
 		}
 
-		txsPoolPerShard[shardID].Txs[string(txHash)] = tx
-		txsHashesPerShard[shardID] = append(txsHashesPerShard[shardID], txHash)
+		tx, ok := txHandler.(*transactionData.Transaction)
+		if !ok {
+			continue
+		}
+		tx.Signature = []byte(common.GenesisTxSignatureString)
+		tx.GasLimit = uint64(0)
+
+		txsPoolPerShard[senderShardID].Txs[string(txHash)] = tx
+		txsPoolPerShard[receiverShardID].Txs[string(txHash)] = tx
+
+		for _, miniBlock := range miniBlocks {
+			if (senderShardID == miniBlock.GetSenderShardID()) &&
+				(receiverShardID == miniBlock.GetReceiverShardID()) {
+				miniBlock.TxHashes = append(miniBlock.TxHashes, txHash)
+			}
+		}
 	}
 
-	miniBlocks := ap.generateIntraShardMiniBlocks(txsHashesPerShard)
+	return nil
+}
+
+// GenerateInitialTransactions will generate initial transactions pool and the miniblocks for the generated transactions
+func (ap *accountsParser) GenerateInitialTransactions(
+	shardCoordinator sharding.Coordinator,
+	indexingData map[uint32]genesis.InitialIndexingDataHandler,
+) ([]*block.MiniBlock, map[uint32]*indexer.Pool, error) {
+	if check.IfNil(shardCoordinator) {
+		return nil, nil, genesis.ErrNilShardCoordinator
+	}
+
+	mintTxs := ap.createMintTransactions()
+	allTxs := ap.getAllTxs(indexingData, mintTxs)
+
+	shardIDs := getShardIDs(shardCoordinator)
+	txsPoolPerShard := ap.createIndexerPools(shardIDs)
+	miniBlocks := createMiniBlocks(shardIDs, block.TxBlock)
+
+	err := ap.setTxsPoolAndMiniBlocks(shardCoordinator, allTxs, txsPoolPerShard, miniBlocks)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	ap.setScrsTxsPool(shardCoordinator, indexingData, txsPoolPerShard)
 
 	return miniBlocks, txsPoolPerShard, nil
 }
+
+// IsInterfaceNil returns if underlying object is true
```

Comment:
`// IsInterfaceNil returns true if the underlying object is nil`

Reply:
:+1: